### ISSUE TITLE
Unable to rename symbol a structure field points to

### ISSUE THREAD

**Issue by joelreymont** [Vector35/binaryninja-api#4240]
**Version and Platform (required):**
- Binary Ninja Version: 3.4.4242-dev, 2e3a4286
- OS: macos
- OS Version: 13.3
- CPU Architecture: arm64

**Bug Description:**

I'm looking at an array of structures. One of the fields is a pointer to a func. I select the func name in the field and pick rename symbol. I enter a new name and nothing happens. It throws me back to the beginning of the array and does nothing visible. It sure does something invisible, though, because Undo works. The same thing works if I click through to the func and rename it there.

This is very annoying as I'm going through an array of 200 structures and clicking through to the func to rename and then ESC to go back returns me to the beginning of the array. Then I have to page down to find the next record.

**Comment by joelreymont** [comment]
It works fine when renaming function names in code.

**Comment by negasora** [comment]
Do you see the updated name in the symbol list and/or does switching ILs show the updated name? It could just be linear view missing an update.

**Comment by joelreymont** [comment]
No, that's not it. The symbol list does not update, clicking through to the function does not show updated name. Switching IL does not show updated name either. It's like the operation does nothing without reporting an error.

**Comment by joelreymont** [comment]
Here's a screenshot. I'm trying to rename by clicking on the contents of the handler field (the sub... values) and pressing 'N'.

![Screenshot 2023-04-19 at 17 15 54](https://user-images.githubusercontent.com/18791/233103536-79162f51-2041-4f50-810d-82160fe8f29d.png)

**Comment by psifertex** [comment]
Hmm, it's not generally broken as I can reproduce the rename working fine here for a function pointer inside an array of structs. I do see that the focus goes to the top of the array when the rename is done which is annoying, but the rename definitely does take effect. Can you DM me the BNDB via slack?

**Comment by joelreymont** [comment]
Same issue trying to rename a function pointer. Apparently, this is a broader issue than above

```
000102ed8  void* data_102ed8 = 0x10003cf
000102edc  void* data_102edc = 0xff00ffff
000102ee0  void* data_102ee0 = sub_b441c
000102ee4  void* data_102ee4 = 0x10003c3
000102ee8  void* data_102ee8 = 0xff00ffff
000102eec  void* data_102eec = sub_b4370
```

**Comment by fuzyll** [comment]
We have the binary above and were able to reproduce the issue. Anyone from V35 should search for "Encourage Salesman Prompt Delay" to find the database.

**Comment by plafosse** [comment]
Root cause of this issue is that we weren't properly masking the thumb bit for these symbol tokens.

Should be fixed in 3.5.4305
